packages/subgraph/tests/dao/dao_v1_0_0.test.ts

```diff
@@ -924,71 +963,102 @@ describe('handleExecuted', () => {
     );
-    let transactionActionsProposalEntityId =
-      generateTransactionActionsProposalEntityId(
-        proposalEntityId,
-        event.transaction.hash,
-        event.transactionLogIndex
-      );
-    let actionEntityId = generateActionEntityId(
-      transactionActionsProposalEntityId,
+
+    let transactionActionsEntityId = generateTransactionActionsEntityId(
+      event.address,
+      event.params.actor,
+      event.params.callId,
+      event.transaction.hash,
+      event.transactionLogIndex
+    );
+
+    let deterministicActionId = generateDeterministicActionId(
+      event.address,
+      event.params.actor,
+      event.params.callId,
       0
     );
+    let actionEntityId = generateTransactionActionEntityId(
+      event.address,
+      event.params.actor,
+      event.params.callId,
+      0,
+      event.transaction.hash,
+      event.transactionLogIndex
+    );
 
     // create proposal
-    let proposal = new TransactionActionsProposal(
-      transactionActionsProposalEntityId
-    );
-    proposal.dao = event.address.toHexString();
-    proposal.createdAt = event.block.timestamp;
-    proposal.endDate = event.block.timestamp;
-    proposal.startDate = event.block.timestamp;
-    proposal.allowFailureMap = BigInt.zero();
-    proposal.creator = event.params.actor;
-    proposal.executionTxHash = event.transaction.hash;
-    proposal.executed = true;
-    proposal.save();
+    let transactionActions = new TransactionActions(transactionActionsEntityId);
+    transactionActions.dao = event.address.toHexString();
+    transactionActions.createdAt = event.block.timestamp;
+    transactionActions.deterministicId = deterministicId;
+    transactionActions.endDate = event.block.timestamp;
+    transactionActions.startDate = event.block.timestamp;
+    transactionActions.allowFailureMap = BigInt.zero();
+    transactionActions.creator = event.params.actor;
+    transactionActions.executionTxHash = event.transaction.hash;
+    transactionActions.executed = true;
+    transactionActions.save();
 
     // create action
-    let action = new Action(actionEntityId);
+    let action = new TransactionAction(actionEntityId);
     action.to = Address.fromString(DAO_TOKEN_ADDRESS);
     action.data = Bytes.fromHexString('0x');
     action.value = BigInt.zero();
     action.dao = event.address.toHexString();
-    action.proposal = proposal.id;
+    action.transactionActions = transactionActions.id;
+    action.deterministicId = deterministicActionId;
     action.save();
 
     // Check that before `handleExecute`, execResults are empty
-    assert.entityCount('Action', 1);
-    assert.entityCount('TransactionActionsProposal', 1);
+    assert.entityCount('TransactionAction', 1);
+    assert.entityCount('TransactionActions', 1);
     assert.assertTrue(action.execResult === null);
-    assert.assertTrue(proposal.failureMap === null);
+    assert.assertTrue(transactionActions.failureMap === null);
 
     handleExecuted(event);
```

Comment:
As I understand this test:
1. It creates `TransactionActions` with a data of the event.
2. Then the same event is emitted. Since the transactionsActionsEntityId will be generated the same as it already exists in the database, why doesn't this cause the fail ? In subgraph, if you're trying to store a record with the same id that already exists, it fails. Why not here ? same applies in dao_v1_3_0.test.s

Reply:
Good question - investigating

Reply:
https://thegraph.com/docs/en/developing/creating-a-subgraph/#handling-of-entities-with-identical-ids

It doesn't fail it just overwrites the entity